### PR TITLE
Added first_account_id getter

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Adds a first_account_id fn for convenient access

### CODE REVIEW

File: sdk/src/wallet/core/mod.rs [hunk 1]
```diff
@@ -313,6 +313,16 @@ impl WalletData {
             .filter(|output_data| output_data.output.is_account())
     }
 
+    // Returns the first possible Account id, which can be an implicit account.
+    // If none are found, returns AccountNotFound.
+    pub fn first_account_id(&self) -> Result<AccountId> {
```
Comment (DaughterOfMars) [inline sdk/src/wallet/core/mod.rs]: Actually, I think this should just return `Option<AccountId>` and the usage should be like

```
let account_id = account_id.or_else(|| wallet_data.first_account_id()).ok_or(Error::AccountNotFound)?;
```

File: cli/src/wallet_cli/mod.rs [hunk 2]
```diff
@@ -356,16 +356,7 @@ pub async fn address_command(wallet: &Wallet) -> Result<(), Error> {
...
-        .or(wallet_data
-            .accounts()
-            .next()
-            .map(|o| o.output.as_account().account_id_non_null(&o.output_id)))
-        .or(wallet_data
-            .implicit_accounts()
-            .next()
-            .map(|o| AccountId::from(&o.output_id)))
-        .ok_or(WalletError::AccountNotFound)?;
+    let account_id = account_id.unwrap_or(wallet_data.first_account_id()?);
```
Comment (Alex6323) [inline cli/src/wallet_cli/mod.rs]: I agree with Chloé, and this here probably also produces a clippy warning, I guess.

File: sdk/src/wallet/core/mod.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/wallet/core/mod.rs]: I agree

**Comment by Alex6323** [comment]
is this PR no longer pursued?

File: cli/src/wallet_cli/mod.rs [hunk 3]
```diff
@@ -1042,8 +1035,7 @@ pub async fn unspent_outputs_command(wallet: &Wallet) -> Result<(), Error> {
 // pub async fn participation_overview_command(
 //     wallet: &Wallet,
 //     event_ids: Option<Vec<ParticipationEventId>>,
-// ) -> Result<(), Error> {
-//     let participation_overview = wallet.get_participation_overview(event_ids).await?;
+// ) -> Result<(), Error> { let participation_overview = wallet.get_participation_overview(event_ids).await?;
```
Comment (kwek20) [inline cli/src/wallet_cli/mod.rs]: hm this was fmt, odd

File: sdk/src/wallet/core/mod.rs [hunk 4]
```diff
@@ -315,12 +315,11 @@ impl WalletData {
 
     // Returns the first possible Account id, which can be an implicit account.
     // If none are found, returns AccountNotFound.
```
Comment (Thoralf-M) [inline sdk/src/wallet/core/mod.rs:317]: Outdated comment

File: sdk/src/wallet/core/mod.rs [hunk 5]
```diff
@@ -313,6 +313,15 @@ impl WalletData {
             .filter(|output_data| output_data.output.is_account())
     }
 
+    // Returns the first possible Account id, which can be an implicit account.
+    // If none are found, returns AccountNotFound.
```
Comment (Thoralf-M) [inline sdk/src/wallet/core/mod.rs]:
```suggestion
```

File: cli/src/wallet_cli/mod.rs [hunk 6]
```diff
@@ -367,14 +367,7 @@ pub async fn address_command(wallet: &Wallet) -> Result<(), Error> {
...
     let account_id = account_id
-        .or(wallet_data
-            .accounts()
-            .next()
-            .map(|o| o.output.as_account().account_id_non_null(&o.output_id)))
-        .or(wallet_data
-            .implicit_accounts()
-            .next()
-            .map(|o| AccountId::from(&o.output_id)))
+        .or_else(|| wallet_data.first_account_id())
```
Comment (thibault-martinez) [inline cli/src/wallet_cli/mod.rs:370]: Also do this in `congestion_command` ?

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]
I don't really like the name but we can revisit later.

**Pull request merged**